turtle_tf2_cpp/src/turtle_tf2_message_filter.cpp

```diff
@@ -0,0 +1,96 @@
+// Copyright 2021 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <rclcpp/rclcpp.hpp>
+#include <geometry_msgs/msg/point_stamped.hpp>
+
+#include <tf2_ros/transform_listener.h>
+#include <tf2_ros/message_filter.h>
+#include <tf2_ros/buffer.h>
+#include <tf2_ros/create_timer_ros.h>
+#ifdef TF2_CPP_HEADERS
+  #include <tf2_geometry_msgs/tf2_geometry_msgs.hpp>
+#else
+  #include <tf2_geometry_msgs/tf2_geometry_msgs.h>
+#endif
+#include <message_filters/subscriber.h>
+
+#include <chrono>
+#include <memory>
+#include <string>
+
+using std::placeholders::_1;
+using namespace std::chrono_literals;
+
+
+class PoseDrawer : public rclcpp::Node
+{
+public:
+  PoseDrawer()
+  : Node("turtle_tf2_pose_drawer")
+  {
+    auto node = rclcpp::Node::make_shared("tf2_ros_message_filter");
+    auto create_timer_interface = std::make_shared<tf2_ros::CreateTimerROS>(
+      node->get_node_base_interface(),
+      node->get_node_timers_interface());
+    typedef std::chrono::duration<int> seconds_type;
+    seconds_type buffer_timeout(1);
+
+    // Declare and acquire `target_frame` parameter
+    this->declare_parameter<std::string>("target_frame", "turtle1");
+    this->get_parameter("target_frame", target_frame_);
+
+    rclcpp::Clock::SharedPtr clock = std::make_shared<rclcpp::Clock>(RCL_SYSTEM_TIME);
+    tf2_buffer_ = std::make_shared<tf2_ros::Buffer>(node->get_clock());
+    tf2_buffer_->setCreateTimerInterface(create_timer_interface);
```

Comment:
Can you please clarify why we need to create `setCreateTimerInterface`? I think a couple of inline comments would suffice

Reply:
This is also because I create another node object except this pointer, and if I didn't setCreateTimerInterface, then when I run this node, it will raise a exception:
terminate called after throwing an instance of 'tf2_ros::CreateTimerInterfaceException'
  what():  timer interface not set before call to waitForTransform
[ros2run]: Aborted
If we can resolve the two Node instance problem, I think we needn't to create setCreateTimerInterface.

Reply:
@kurshakuz Still need setCreateTimerInterface after changing from two node instances to one node instance, and I added two inline comments for its purpose.